packages/react-meteor-data/suspense/useFind.ts

```diff
@@ -0,0 +1,126 @@
+import { Meteor } from 'meteor/meteor'
+import { Mongo } from 'meteor/mongo'
+import { useEffect } from 'react'
+import isEqual from 'lodash/isEqual';
+import clone from 'lodash/clone'
+
+
+export const selectorsCache = new Map<Mongo.Collection<unknown>, SelectorEntry[]>();
+type SelectorEntry = {
+  findArgs: Parameters<Mongo.Collection<unknown>['find']>;
+  promise: Promise<unknown>;
+  result?: unknown;
+  error?: unknown;
+  counter: number;
+
+};
+export const removeFromArray =
+  <T>(list: T[], obj: T): void => {
+    if (obj) {
+      const index = list.indexOf(obj);
+      if (index !== -1) {
+        list.splice(index, 1);
+      }
+    }
+  }
+const useFindSuspense = <T = any>(
+  collection: Mongo.Collection<T>,
+  findArgs: Parameters<Mongo.Collection<T>['find']> | null,
+) => {
+  useEffect(() => {
+    const cachedSelectors = selectorsCache.get(collection) ?? [];
+    const _args = clone(findArgs)
```

Comment:
Do we need to `clone` it? I remember we did that during testing, but I don't think it's needed.